### ISSUE TITLE
Action always passes without detecting secrets

### ISSUE THREAD

**Issue by sammcj** [edplato/trufflehog-actions-scan#17]
When I try to use this action, it completes without error but does not find any secrets (regex or high entropy).

I have:

- Ensured I'm using checkout@v2 with `fetch-depth: "0"` before the trufflehog step.
- Confirmed with a `ls -laR` that the git repo is indeed checked out in the PWD.
- Explicitly specifying `--rules`, `--entropy` `--max_depth`.
- Running trufflehog locally in the same git repo and it works as expected (detects secrets).

Configuration:

```
  TruffleHog:
    runs-on: [self-hosted]
    steps:
    - uses: actions/checkout@v2
      with:
        fetch-depth: "0"
    - run: ls -laR
    - uses: edplato/trufflehog-actions-scan@master
      with:
        scanArguments: "--regex --entropy=true --max_depth=1000 --rules regexes.json"
```

Output from Action with a repo that contains some fake secrets:

_Note that it doesn't look like the action is even running trufflehog_

![Screen Shot 2021-12-15 at 10 53 18 am](https://user-images.githubusercontent.com/862951/146097815-e68703d1-fb84-4e57-a0b6-e725d0e0d7a4.jpg)

Example output from running Trufflehog locally with the same repo:

```
trufflehog --regex --entropy=false --max_depth=1000 --rules regexes.json file://$(pwd)

-----BEGIN EC PRIVATE KEY-----
~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~~~~~~
Reason: PGP private key block
Date: 2021-12-09 10:43:49
Hash: 87b860ae90f535ebc11646e9753be23894844966
Filepath: .gitleaks.toml
Branch: origin/secrets_scanning
Commit: add mvp of trufflehog secrets scanning

-----BEGIN PGP PRIVATE KEY BLOCK-----
~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~~~~~~
Reason: Generic Secret
Date: 2021-12-09 10:43:49
Hash: 87b860ae90f535ebc11646e9753be23894844966
Filepath: fakesecretitems/testbadcode.py
Branch: origin/secrets_scanning
Commit: add mvp of trufflehog secrets scanning

secret : "lolasifidsaveapasswordinplaintextbro"
```

**Comment by edplato** [comment]
It appears to be running - there would an error if not. Out of curiosity, expand here to see those hidden lines to see if your arguments are included:
![image](https://user-images.githubusercontent.com/24417194/146404063-85594a0d-530c-4138-9ac2-0e0032bf45c6.png)

My first thought is using `actions/checkout@v2` with `[self-hosted]` may be causing the issue. Could you try with `ubuntu-latest` and see if the problem persists?

**Comment by sammcj** [comment]
Sorry I should have stated that I'd tried on `ubuntu-latest` as well, it has the same behaviour.

Here is the expanded run step:
![Screen Shot 2021-12-17 at 8 43 05 am](https://user-images.githubusercontent.com/862951/146453345-b7abfdaa-28f5-4c60-9aa4-af7b739ab1ab.jpg)


And the full job output:

```
2021-12-16T21:44:06.3873050Z Found online and idle hosted runner in the current repository's enterprise account that matches the required labels: 'ubuntu-latest'
2021-12-16T21:44:06.4372604Z Waiting for a Hosted runner in the 'enterprise' to pick this job...
2021-12-16T21:44:06.8199066Z Job is waiting for a hosted runner to come online.
2021-12-16T21:44:10.2955744Z Job is about to start running on the hosted runner: GitHub Actions 13 (hosted)
2021-12-16T21:44:14.5254121Z Current runner version: '2.285.1'
2021-12-16T21:44:14.5286272Z ##[group]Operating System
2021-12-16T21:44:14.5287322Z Ubuntu
2021-12-16T21:44:14.5287917Z 20.04.3
2021-12-16T21:44:14.5288400Z LTS
2021-12-16T21:44:14.5288974Z ##[endgroup]
2021-12-16T21:44:14.5289603Z ##[group]Virtual Environment
2021-12-16T21:44:14.5290397Z Environment: ubuntu-20.04
2021-12-16T21:44:14.5291093Z Version: 20211214.2
2021-12-16T21:44:14.5292234Z Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20211214.2/images/linux/Ubuntu2004-README.md
2021-12-16T21:44:14.5293860Z Image Release: https://github.com/actions/virtual-environments/releases/tag/ubuntu20%2F20211214.2
2021-12-16T21:44:14.5294919Z ##[endgroup]
2021-12-16T21:44:14.5295612Z ##[group]Virtual Environment Provisioner
2021-12-16T21:44:14.5296403Z 1.0.0.0-main-20211208-1
2021-12-16T21:44:14.5296979Z ##[endgroup]
2021-12-16T21:44:14.5298495Z ##[group]GITHUB_TOKEN Permissions
2021-12-16T21:44:14.5299785Z Contents: read
2021-12-16T21:44:14.5300549Z Metadata: read
2021-12-16T21:44:14.5301304Z ##[endgroup]
2021-12-16T21:44:14.5304506Z Secret source: Actions
2021-12-16T21:44:14.5305581Z Prepare workflow directory
2021-12-16T21:44:14.6011525Z Prepare all required actions
2021-12-16T21:44:14.6023330Z Getting action download info
2021-12-16T21:44:14.7857813Z Download action repository 'actions/checkout@v2' (SHA:ec3a7ce113134d7a93b817d10a8272cb61118579)
2021-12-16T21:44:16.4717204Z Download action repository 'edplato/trufflehog-actions-scan@master' (SHA:c36ff9abf0af8290ef23b1b45a36e75c742dd1d8)
2021-12-16T21:44:16.8681852Z ##[group]Build container for action use: '/home/runner/work/_actions/edplato/trufflehog-actions-scan/master/Dockerfile'.
2021-12-16T21:44:16.8744269Z ##[command]/usr/bin/docker build -t 6a6825:c9be214cb0eb44408199c4b04e6a1d43 -f "/home/runner/work/_actions/edplato/trufflehog-actions-scan/master/Dockerfile" "/home/runner/work/_actions/edplato/trufflehog-actions-scan/master"
2021-12-16T21:44:17.1465828Z Sending build context to Docker daemon  14.34kB
2021-12-16T21:44:17.1485004Z 
2021-12-16T21:44:17.1907225Z Step 1/18 : FROM python:3.7-alpine
2021-12-16T21:44:17.4271733Z 3.7-alpine: Pulling from library/python
2021-12-16T21:44:17.4845659Z 59bf1c3509f3: Pulling fs layer
2021-12-16T21:44:17.4846735Z 8786870f2876: Pulling fs layer
2021-12-16T21:44:17.4849835Z 45d4696938d0: Pulling fs layer
2021-12-16T21:44:17.4851728Z ef84af58b2c5: Pulling fs layer
2021-12-16T21:44:17.4852743Z c3c9b71b9a69: Pulling fs layer
2021-12-16T21:44:17.4859134Z ef84af58b2c5: Waiting
2021-12-16T21:44:17.4860581Z c3c9b71b9a69: Waiting
2021-12-16T21:44:17.5621971Z 8786870f2876: Verifying Checksum
2021-12-16T21:44:17.5622635Z 8786870f2876: Download complete
2021-12-16T21:44:17.5875566Z 59bf1c3509f3: Verifying Checksum
2021-12-16T21:44:17.5878354Z 59bf1c3509f3: Download complete
2021-12-16T21:44:18.4735770Z 45d4696938d0: Verifying Checksum
2021-12-16T21:44:18.4751999Z 45d4696938d0: Download complete
2021-12-16T21:44:18.4752953Z ef84af58b2c5: Verifying Checksum
2021-12-16T21:44:18.4753788Z ef84af58b2c5: Download complete
2021-12-16T21:44:18.4754550Z c3c9b71b9a69: Verifying Checksum
2021-12-16T21:44:18.4755430Z c3c9b71b9a69: Download complete
2021-12-16T21:44:18.4756147Z 59bf1c3509f3: Pull complete
2021-12-16T21:44:18.4756791Z 8786870f2876: Pull complete
2021-12-16T21:44:19.0122712Z 45d4696938d0: Pull complete
2021-12-16T21:44:19.0947782Z ef84af58b2c5: Pull complete
2021-12-16T21:44:19.3944027Z c3c9b71b9a69: Pull complete
2021-12-16T21:44:19.4052145Z Digest: sha256:d64e0124674d64e78cc9d7378a1130499ced66a7a00db0521d0120a2e88ac9e4
2021-12-16T21:44:19.4084783Z Status: Downloaded newer image for python:3.7-alpine
2021-12-16T21:44:19.4108050Z  ---> a1034fd13493
2021-12-16T21:44:19.4108807Z Step 2/18 : ENV HTTP_PROXY=$HTTP_PROXY
2021-12-16T21:44:19.4428425Z  ---> Running in f3e35dc1fdb9
2021-12-16T21:44:20.0242386Z Removing intermediate container f3e35dc1fdb9
2021-12-16T21:44:20.0243289Z  ---> f2fa7083bea0
2021-12-16T21:44:20.0323838Z Step 3/18 : ENV HTTPS_PROXY=$HTTPS_PROXY
2021-12-16T21:44:20.0569957Z  ---> Running in a80251bc8c44
2021-12-16T21:44:21.0177023Z Removing intermediate container a80251bc8c44
2021-12-16T21:44:21.0179645Z  ---> 1b14f1f2d00a
2021-12-16T21:44:21.0202835Z Step 4/18 : ENV NO_PROXY=$NO_PROXY
2021-12-16T21:44:21.0470323Z  ---> Running in fa8e4f9bab05
2021-12-16T21:44:22.0197192Z Removing intermediate container fa8e4f9bab05
2021-12-16T21:44:22.0197975Z  ---> 27a4303631d0
2021-12-16T21:44:22.0198868Z Step 5/18 : LABEL name="trufflehog-actions-scan"
2021-12-16T21:44:22.0433251Z  ---> Running in 309a1348b9c6
2021-12-16T21:44:23.0168924Z Removing intermediate container 309a1348b9c6
2021-12-16T21:44:23.0185101Z  ---> 0e0d2bd593ba
2021-12-16T21:44:23.0187635Z Step 6/18 : LABEL version="1.0.0"
2021-12-16T21:44:23.0467687Z  ---> Running in 93af5cf96608
2021-12-16T21:44:24.0194863Z Removing intermediate container 93af5cf96608
2021-12-16T21:44:24.0195755Z  ---> f6d5e01f9df3
2021-12-16T21:44:24.0196779Z Step 7/18 : LABEL repository="https://github.com/edplato/trufflehog-actions-scan"
2021-12-16T21:44:24.0529254Z  ---> Running in d9a2c2b7b892
2021-12-16T21:44:25.0171323Z Removing intermediate container d9a2c2b7b892
2021-12-16T21:44:25.0172344Z  ---> b40de251f58c
2021-12-16T21:44:25.0173273Z Step 8/18 : LABEL homepage="https://github.com/edplato/trufflehog-actions-scan"
2021-12-16T21:44:25.0502781Z  ---> Running in 2c58f5135dad
2021-12-16T21:44:26.0241579Z Removing intermediate container 2c58f5135dad
2021-12-16T21:44:26.0242659Z  ---> be658254a0b3
2021-12-16T21:44:26.0243447Z Step 9/18 : LABEL maintainer="Ed Plato"
2021-12-16T21:44:26.0505191Z  ---> Running in 3a7ac5d67da8
2021-12-16T21:44:27.0194627Z Removing intermediate container 3a7ac5d67da8
2021-12-16T21:44:27.0195310Z  ---> 62dae9a8db98
2021-12-16T21:44:27.0196090Z Step 10/18 : LABEL "com.github.actions.name"="Trufflehog Actions Scan"
2021-12-16T21:44:27.0447665Z  ---> Running in 603722e96ea9
2021-12-16T21:44:28.0175103Z Removing intermediate container 603722e96ea9
2021-12-16T21:44:28.0175751Z  ---> 6a1b1c26db32
2021-12-16T21:44:28.0176921Z Step 11/18 : LABEL "com.github.actions.description"="Scan repository for secrets with basic trufflehog defaults in place for easy setup."
2021-12-16T21:44:28.0391066Z  ---> Running in a7645b9107da
2021-12-16T21:44:29.0145314Z Removing intermediate container a7645b9107da
2021-12-16T21:44:29.0146024Z  ---> aa06762745c3
2021-12-16T21:44:29.0146716Z Step 12/18 : LABEL "com.github.actions.icon"="shield"
2021-12-16T21:44:29.0381959Z  ---> Running in 268b8bff609e
2021-12-16T21:44:30.0195310Z Removing intermediate container 268b8bff609e
2021-12-16T21:44:30.0195998Z  ---> ce93b48506cc
2021-12-16T21:44:30.0196713Z Step 13/18 : LABEL "com.github.actions.color"="yellow"
2021-12-16T21:44:30.0406855Z  ---> Running in 906026c68424
2021-12-16T21:44:31.0146983Z Removing intermediate container 906026c68424
2021-12-16T21:44:31.0147692Z  ---> 8dd1715effa6
2021-12-16T21:44:31.0148263Z Step 14/18 : RUN pip3 install truffleHog==2.2.1
2021-12-16T21:44:31.0393843Z  ---> Running in 495286a56934
2021-12-16T21:44:33.2516385Z Collecting truffleHog==2.2.1
2021-12-16T21:44:33.3067889Z   Downloading truffleHog-2.2.1-py2.py3-none-any.whl (13 kB)
2021-12-16T21:44:33.3324257Z Collecting truffleHogRegexes==0.0.7
2021-12-16T21:44:33.3476336Z   Downloading truffleHogRegexes-0.0.7-py2.py3-none-any.whl (2.5 kB)
2021-12-16T21:44:33.4159513Z Collecting GitPython==3.0.6
2021-12-16T21:44:33.4296667Z   Downloading GitPython-3.0.6-py3-none-any.whl (451 kB)
2021-12-16T21:44:33.4911566Z Collecting gitdb2>=2.0.0
2021-12-16T21:44:33.5022305Z   Downloading gitdb2-4.0.2-py3-none-any.whl (1.1 kB)
2021-12-16T21:44:33.5281982Z Collecting gitdb>=4.0.1
2021-12-16T21:44:33.5396951Z   Downloading gitdb-4.0.9-py3-none-any.whl (63 kB)
2021-12-16T21:44:33.5794708Z Collecting smmap<6,>=3.0.1
2021-12-16T21:44:33.5926566Z   Downloading smmap-5.0.0-py3-none-any.whl (24 kB)
2021-12-16T21:44:33.6526223Z Installing collected packages: smmap, gitdb, gitdb2, truffleHogRegexes, GitPython, truffleHog
2021-12-16T21:44:33.9888129Z [91mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
2021-12-16T21:44:33.9892241Z [0mSuccessfully installed GitPython-3.0.6 gitdb-4.0.9 gitdb2-4.0.2 smmap-5.0.0 truffleHog-2.2.1 truffleHogRegexes-0.0.7
2021-12-16T21:44:34.1105760Z [91mWARNING: You are using pip version 21.2.4; however, version 21.3.1 is available.
2021-12-16T21:44:34.1107683Z You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.
2021-12-16T21:44:35.0421723Z [0mRemoving intermediate container 495286a56934
2021-12-16T21:44:35.0422414Z  ---> 84a6d3047107
2021-12-16T21:44:35.0423159Z Step 15/18 : RUN apk --update add git less openssh &&   rm -rf /var/lib/apt/lists/* &&   rm /var/cache/apk/*
2021-12-16T21:44:35.0682042Z  ---> Running in 6759be64a0ab
2021-12-16T21:44:35.3616904Z fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
2021-12-16T21:44:35.5361768Z fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
2021-12-16T21:44:35.8874040Z (1/14) Installing brotli-libs (1.0.9-r5)
2021-12-16T21:44:35.9106775Z (2/14) Installing nghttp2-libs (1.46.0-r0)
2021-12-16T21:44:35.9218933Z (3/14) Installing libcurl (7.80.0-r0)
2021-12-16T21:44:35.9410685Z (4/14) Installing pcre2 (10.39-r0)
2021-12-16T21:44:35.9648358Z (5/14) Installing git (2.34.1-r0)
2021-12-16T21:44:36.1439020Z (6/14) Installing less (590-r0)
2021-12-16T21:44:36.1558819Z (7/14) Installing openssh-keygen (8.8_p1-r1)
2021-12-16T21:44:36.1712126Z (8/14) Installing libedit (20210910.3.1-r0)
2021-12-16T21:44:36.1831087Z (9/14) Installing openssh-client-common (8.8_p1-r1)
2021-12-16T21:44:36.2172861Z (10/14) Installing openssh-client-default (8.8_p1-r1)
2021-12-16T21:44:36.2362889Z (11/14) Installing openssh-sftp-server (8.8_p1-r1)
2021-12-16T21:44:36.2462233Z (12/14) Installing openssh-server-common (8.8_p1-r1)
2021-12-16T21:44:36.2544791Z (13/14) Installing openssh-server (8.8_p1-r1)
2021-12-16T21:44:36.2738010Z (14/14) Installing openssh (8.8_p1-r1)
2021-12-16T21:44:36.2904995Z Executing busybox-1.34.1-r3.trigger
2021-12-16T21:44:36.2958432Z OK: 30 MiB in 49 packages
2021-12-16T21:44:37.0249941Z Removing intermediate container 6759be64a0ab
2021-12-16T21:44:37.0250626Z  ---> 9b9eed68508e
2021-12-16T21:44:37.0251241Z Step 16/18 : ADD entrypoint.sh  /entrypoint.sh
2021-12-16T21:44:38.0177657Z  ---> 600405735cd4
2021-12-16T21:44:38.0178341Z Step 17/18 : ADD regexes.json /regexes.json
2021-12-16T21:44:39.0184613Z  ---> 4bc926ab6b8f
2021-12-16T21:44:39.0186883Z Step 18/18 : ENTRYPOINT ["/entrypoint.sh"]
2021-12-16T21:44:39.0431871Z  ---> Running in 9b3fc5e170f0
2021-12-16T21:44:40.0197472Z Removing intermediate container 9b3fc5e170f0
2021-12-16T21:44:40.0212861Z  ---> 4bda55af56aa
2021-12-16T21:44:40.0213663Z Successfully built 4bda55af56aa
2021-12-16T21:44:40.0269809Z Successfully tagged 6a6825:c9be214cb0eb44408199c4b04e6a1d43
2021-12-16T21:44:40.0302463Z ##[endgroup]
2021-12-16T21:44:40.0646893Z ##[group]Run actions/checkout@v2
2021-12-16T21:44:40.0647553Z with:
2021-12-16T21:44:40.0648004Z   fetch-depth: 0
2021-12-16T21:44:40.0648648Z   repository: <redacetd>/jayp-app-github
2021-12-16T21:44:40.0649895Z   token: ***
2021-12-16T21:44:40.0650355Z   ssh-strict: true
2021-12-16T21:44:40.0650932Z   persist-credentials: true
2021-12-16T21:44:40.0651498Z   clean: true
2021-12-16T21:44:40.0651903Z   lfs: false
2021-12-16T21:44:40.0652359Z   submodules: false
2021-12-16T21:44:40.0652822Z ##[endgroup]
2021-12-16T21:44:40.3778905Z Syncing repository: reecegroup/jayp-app-github
2021-12-16T21:44:40.3780496Z ##[group]Getting Git version info
2021-12-16T21:44:40.3781640Z Working directory is '/home/runner/work/jayp-app-github/jayp-app-github'
2021-12-16T21:44:40.3797333Z [command]/usr/bin/git version
2021-12-16T21:44:40.3957288Z git version 2.34.1
2021-12-16T21:44:40.3984205Z ##[endgroup]
2021-12-16T21:44:40.3993346Z Deleting the contents of '/home/runner/work/jayp-app-github/jayp-app-github'
2021-12-16T21:44:40.4001124Z ##[group]Initializing the repository
2021-12-16T21:44:40.4006796Z [command]/usr/bin/git init /home/runner/work/jayp-app-github/jayp-app-github
2021-12-16T21:44:40.4100536Z hint: Using 'master' as the name for the initial branch. This default branch name
2021-12-16T21:44:40.4102010Z hint: is subject to change. To configure the initial branch name to use in all
2021-12-16T21:44:40.4103172Z hint: of your new repositories, which will suppress this warning, call:
2021-12-16T21:44:40.4104055Z hint: 
2021-12-16T21:44:40.4105760Z hint: 	git config --global init.defaultBranch <name>
2021-12-16T21:44:40.4106468Z hint: 
2021-12-16T21:44:40.4107338Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2021-12-16T21:44:40.4108732Z hint: 'development'. The just-created branch can be renamed via this command:
2021-12-16T21:44:40.4109560Z hint: 
2021-12-16T21:44:40.4110170Z hint: 	git branch -m <name>
2021-12-16T21:44:40.4114340Z Initialized empty Git repository in /home/runner/work/jayp-app-github/jayp-app-github/.git/
2021-12-16T21:44:40.4127621Z [command]/usr/bin/git remote add origin https://github.com/reecegroup/jayp-app-github
2021-12-16T21:44:40.4178870Z ##[endgroup]
2021-12-16T21:44:40.4180301Z ##[group]Disabling automatic garbage collection
2021-12-16T21:44:40.4188595Z [command]/usr/bin/git config --local gc.auto 0
2021-12-16T21:44:40.4234216Z ##[endgroup]
2021-12-16T21:44:40.4245912Z ##[group]Setting up auth
2021-12-16T21:44:40.4257416Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2021-12-16T21:44:40.4297478Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2021-12-16T21:44:40.4751923Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2021-12-16T21:44:40.4837792Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2021-12-16T21:44:40.5140764Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2021-12-16T21:44:40.5191397Z ##[endgroup]
2021-12-16T21:44:40.5193028Z ##[group]Fetching the repository
2021-12-16T21:44:40.5204882Z [command]/usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/* +fdc0fc2f87bc8b2eb815286c18839515bf5ab7b3:refs/remotes/pull/8/merge
2021-12-16T21:44:40.9232363Z remote: Enumerating objects: 611, done.        
2021-12-16T21:44:40.9285884Z remote: Counting objects:   0% (1/611)        
2021-12-16T21:44:40.9287526Z remote: Counting objects:   1% (7/611)        
2021-12-16T21:44:40.9290961Z remote: Counting objects:   2% (13/611)        
... 
2021-12-16T21:44:40.9382028Z remote: Counting objects: 100% (611/611)        
2021-12-16T21:44:40.9382768Z remote: Counting objects: 100% (611/611), done.        
2021-12-16T21:44:40.9383542Z remote: Compressing objects:   0% (1/362)        
2021-12-16T21:44:40.9384239Z remote: Compressing objects:   1% (4/362)        
2021-12-16T21:44:40.9384987Z remote: Compressing objects:   2% (8/362)             
...  
2021-12-16T21:44:40.9516405Z remote: Compressing objects: 100% (362/362)        
2021-12-16T21:44:40.9517246Z remote: Compressing objects: 100% (362/362), done.        
2021-12-16T21:44:40.9518057Z Receiving objects:   0% (1/611)
2021-12-16T21:44:40.9518640Z Receiving objects:   1% (7/611)
2021-12-16T21:44:40.9519362Z Receiving objects:   2% (13/611)
...
2021-12-16T21:44:40.9751042Z Receiving objects:  98% (599/611)
2021-12-16T21:44:40.9751579Z Receiving objects:  99% (605/611)
2021-12-16T21:44:40.9752134Z Receiving objects: 100% (611/611)
2021-12-16T21:44:40.9752768Z Receiving objects: 100% (611/611), 106.33 KiB | 5.60 MiB/s, done.
2021-12-16T21:44:40.9753380Z Resolving deltas:   0% (0/267)
2021-12-16T21:44:40.9753924Z Resolving deltas:   1% (3/267)
2021-12-16T21:44:40.9754473Z Resolving deltas:   2% (6/267)
2021-12-16T21:44:40.9755007Z Resolving deltas:   3% (9/267)
...
2021-12-16T21:44:40.9806487Z Resolving deltas:  98% (262/267)
2021-12-16T21:44:40.9807022Z Resolving deltas:  99% (265/267)
2021-12-16T21:44:40.9807542Z Resolving deltas: 100% (267/267)
2021-12-16T21:44:40.9808200Z Resolving deltas: 100% (267/267), done.
2021-12-16T21:44:40.9843491Z From https://github.com/reecegroup/jayp-app-github
...
2021-12-16T21:44:40.9949424Z ##[group]Determining the checkout info
2021-12-16T21:44:40.9950889Z ##[endgroup]
2021-12-16T21:44:40.9952212Z ##[group]Checking out the ref
2021-12-16T21:44:40.9957293Z [command]/usr/bin/git checkout --progress --force refs/remotes/pull/8/merge
2021-12-16T21:44:41.0029990Z Note: switching to 'refs/remotes/pull/8/merge'.
2021-12-16T21:44:41.0030793Z 
2021-12-16T21:44:41.0032465Z You are in 'detached HEAD' state. You can look around, make experimental
2021-12-16T21:44:41.0033434Z changes and commit them, and you can discard any commits you make in this
2021-12-16T21:44:41.0034343Z state without impacting any branches by switching back to a branch.
2021-12-16T21:44:41.0034860Z 
2021-12-16T21:44:41.0035482Z If you want to create a new branch to retain commits you create, you may
2021-12-16T21:44:41.0036553Z do so (now or later) by using -c with the switch command. Example:
2021-12-16T21:44:41.0037024Z 
2021-12-16T21:44:41.0037696Z   git switch -c <new-branch-name>
2021-12-16T21:44:41.0038121Z 
2021-12-16T21:44:41.0038573Z Or undo this operation with:
2021-12-16T21:44:41.0038934Z 
2021-12-16T21:44:41.0039466Z   git switch -
2021-12-16T21:44:41.0039751Z 
2021-12-16T21:44:41.0040478Z Turn off this advice by setting config variable advice.detachedHead to false
2021-12-16T21:44:41.0041284Z 
2021-12-16T21:44:41.0042249Z HEAD is now at fdc0fc2 Merge 54b74295d5a68d0897ddde088a6247cf4d69bf0f into 1819cea130e02f055d8d4e33fd277f7fec749dc3
2021-12-16T21:44:41.0046137Z ##[endgroup]
2021-12-16T21:44:41.0105950Z [command]/usr/bin/git log -1 --format='%H'
2021-12-16T21:44:41.0142067Z 'fdc0fc2f87bc8b2eb815286c18839515bf5ab7b3'
2021-12-16T21:44:41.0298878Z ##[group]Run ls -laR
2021-12-16T21:44:41.0299400Z [36;1mls -laR[0m
2021-12-16T21:44:41.0349147Z shell: /usr/bin/bash -e {0}
2021-12-16T21:44:41.0349660Z ##[endgroup]
2021-12-16T21:44:41.0454734Z .:
2021-12-16T21:44:41.0455242Z total 128
2021-12-16T21:44:41.0456299Z drwxr-xr-x 12 runner docker  4096 Dec 16 21:44 .
2021-12-16T21:44:41.0457155Z drwxr-xr-x  3 runner docker  4096 Dec 16 21:44 ..
2021-12-16T21:44:41.0458042Z drwxr-xr-x  8 runner docker  4096 Dec 16 21:44 .git
2021-12-16T21:44:41.0458927Z drwxr-xr-x  3 runner docker  4096 Dec 16 21:44 .github
2021-12-16T21:44:41.0459822Z -rw-r--r--  1 runner docker  1740 Dec 16 21:44 .gitignore
2021-12-16T21:44:41.0460785Z -rwxr-xr-x  1 runner docker 16641 Dec 16 21:44 .gitleaks.toml
...
```

Note I'm adding a number of fake secrets that match the regexes for Trufflehog running locally here:
```
2021-12-16T21:44:41.0652862Z ./fakesecretitems:
2021-12-16T21:44:41.0653350Z total 20
2021-12-16T21:44:41.0654032Z drwxr-xr-x  2 runner docker 4096 Dec 16 21:44 .
2021-12-16T21:44:41.0654859Z drwxr-xr-x 12 runner docker 4096 Dec 16 21:44 ..
2021-12-16T21:44:41.0655771Z -rw-r--r--  1 runner docker   53 Dec 16 21:44 testbadcode.py
2021-12-16T21:44:41.0656785Z -rw-r--r--  1 runner docker   45 Dec 16 21:44 testbadfile.ini
2021-12-16T21:44:41.0657789Z -rw-r--r--  1 runner docker  516 Dec 16 21:44 testbadkey.pem
```

```
2021-12-16T21:44:41.0783365Z ##[group]Run edplato/trufflehog-actions-scan@master
2021-12-16T21:44:41.0784088Z with:
2021-12-16T21:44:41.0784808Z   scanArguments: --regex --entropy=true --max_depth=1000 --rules regexes.json
2021-12-16T21:44:41.0785567Z ##[endgroup]

2021-12-16T21:44:41.0831372Z ##[command]/usr/bin/docker run --name a6825c9be214cb0eb44408199c4b04e6a1d43_7a088c --label 6a6825 --workdir /github/workspace --rm -e INPUT_SCANARGUMENTS -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/jayp-app-github/jayp-app-github":"/github/workspace" 6a6825:c9be214cb0eb44408199c4b04e6a1d43

**Comment by edplato** [comment]
I am unable to reproduce this in a few different scenarios and repositories. Have you tried only running the action with the recommended default steps to check the result?

Unrelated, but caught my eye in your logs:
![image](https://user-images.githubusercontent.com/24417194/146620097-0388600e-642e-4e0c-8011-dea81d6e2793.png)

**Comment by sammcj** [comment]
I did try the example job definition first and it didn't work.

I think you might be onto something with those weird characters / the repo itself - I just ran the exact same job (copy/paste even) onto another repo that's very similar and it works every time!

I'm sorry to waste your time with this, I suspect it is a repo issue, if I figure out what it is I'll post an update here incase anyone has a similar issue in the future.
